libs/src/services/creatorNode/index.js

```diff
@@ -435,6 +448,60 @@ class CreatorNode {
     this.authToken = null
   }
 
+  /**
+   * Gets and returns the clock values across the replica set for the wallet in userStateManager.
+   * @returns {Object[]} Array of objects with the structure:
+   *
+   * {
+   *  type: 'primary' or 'secondary',
+   *  endpoint: <Content Node endpoint>,
+   *  clockValue: clock value (should be an integer) or null
+   * }
+   *
+   * 'clockValue' may be null if the request to fetch the clock value fails
+   */
+  async getClockValuesFromReplicaSet () {
+    const user = this.userStateManager.getCurrentUser()
+    if (!user || !user.creator_node_endpoint) {
+      console.error('No user or Content Node endpoint found')
+      return
+    }
+
+    const replicaSet = CreatorNode.getEndpoints(user.creator_node_endpoint)
+    const clockValueResponses = await Promise.all(
```

Comment:
i think promise.all will error if any of the promises in it throw an error. we may need to handle the case where one of the replica set values errors

Reply:
maybe something like https://stackoverflow.com/a/36115549/11240376 
`Promise.all([a, b, c].map(p => p.catch(e => e)))`

Reply:
I specifically wrapped `getClockValue()` in the private method `_clockValueRequest()` to avoid any thrown errors :) 

https://github.com/AudiusProject/audius-protocol/pull/1126/files#diff-b1e49d4c865a8d40686e05f6ac646d90ecdbd5e882f6179f5ceffbc5a549af04R471

Reply:
@vicky-g lets say 1 out of the 3 requests fail, the entire Promise.all fails and no values are returned. is that the intention here?

Reply:
we could also use something like this https://github.com/AudiusProject/audius-protocol/blob/d371c76b202babc4281cca040469b177c303a649/libs/src/utils/network.js#L160

Reply:
resolved offline